app/scenes/Document/components/Document.js

```diff
@@ -220,6 +222,12 @@ class DocumentScene extends React.Component<Props> {
     document.title = title;
     document.text = text;
 
+    if (options.collectionId) {
```

Comment:
We need to check authorization to publish into the collection and parent document here, `create` endpoint should have the needed logic given it was responsible for this before